File: app/src/main/java/com/example/readative/room/entity/Author.java
package com.example.readative.room.entity;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity
public class Author {
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "author_id")
    public long id;

    @NonNull
    @ColumnInfo(name = "author_first_name")
    public String firstName;
    @ColumnInfo(name = "author_middle_name")
    public String middleName;
    @ColumnInfo(name = "author_last_name")
    public String lastName;
}

File: app/src/main/java/com/example/readative/feature_book/data/data_source/models/Author.kt
package com.example.readative.feature_book.data.data_source.models

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "authors")
data class Author (
    @PrimaryKey(autoGenerate = true) @ColumnInfo(name = "id") override val id: Long,
    @ColumnInfo(name = "first_name") val firstName: String,
    @ColumnInfo(name = "middle_name") val middleName: String?,
    @ColumnInfo(name = "last_name") val lastName: String?
) : ReadativeEntity